portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadminuser/MakeARelationshipWithSystemObjectThroughAPI.testcase

```diff
@@ -0,0 +1,411 @@
+@component-name = "portal-headless"
+definition {
+
+	property custom.properties = "feature.flag.LPS-162964=true${line.separator}feature.flag.LPS-153324=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless";
+
+	setUp {
+		TestCase.setUpPortalInstanceNoSelenium();
+
+		task ("Given a Foundation object") {
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_label = "foundation",
+				en_US_plural_label = "foundations",
+				name = "Foundation",
+				requiredStringFieldName = "name");
+		}
+	}
+
+	tearDown {
+		UserAccountAPI.deleteRelationship(name = "User");
+
+		for (var objectName : list "Foundation,SecondFoundation") {
+			ObjectAdmin.deleteObjectViaAPI(objectName = "${objectName}");
+		}
+
+		for (var userEmailAddress : list "user1@liferay.com,user2@liferay.com") {
+			JSONUser.deleteUserByEmailAddress(userEmailAddress = "${userEmailAddress}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "4"
+	test CanDeleteCustomObjectEntryRelatedToSystemObjectByManyToMany {
+		property portal.acceptance = "true";
+
+		task ("Given manyToMany relationship usersFoundations created") {
```

Comment:
there are still a lot of Given steps that can be moved all together to the setUp.